numpy/lib/function_base.py

```diff
@@ -1522,17 +1531,81 @@ def gradient(f, *varargs, **kwargs):
     -------
     gradient : ndarray or list of ndarray
         A set of ndarrays (or a single ndarray if there is only one dimension)
-        correposnding to the derivatives of f with respect to each dimension.
+        corresponding to the derivatives of f with respect to each dimension.
         Each derivative has the same shape as f.
 
+    Notes
```

Comment:
I would move this below the examples, just because I suspect examples are more relevant to users most of the time.